rp2040-hal/src/rtc/mod.rs

```diff
@@ -29,8 +29,6 @@ mod filter;
 
 pub use self::filter::DateTimeFilter;
 
-#[cfg_attr(feature = "chrono", path = "datetime_chrono.rs")]
-#[cfg_attr(not(feature = "chrono"), path = "datetime_no_deps.rs")]
```

Comment:
For the motivation of that change in this PR, this feature currently changes the shape of the API.

This makes the rtc examples require conditional compilation to accommodate with the different DateTime creation style.

Following the philosophy of features being additive, it seems more natural that enabling the `chrono` feature _adds_ chrono support rather than _replace_ the local types.

Reply:
That's a very good point for making the no-deps implementation available unconditionally. But in it's current form, this PR completely removes chrono support. (It leaves the `datetime_chrono.rs` file in place, but there's no longer any code referencing it.)

Reply:
[This block](https://github.com/rp-rs/rp-hal/pull/676/files#diff-c111db841a5d3011d4c1f34c3ab3b11cb01d38f58a5d9484807a8a80d7949d5bR136-R207) should be enough to allow easy conversion between chrono's `NaiveDateTime` and our own `DateTime` type.

Reply:
@jonathanpallant recommended using `chrono` in the previous PR #207, which is why it was in #213